tests/io_components/test_utils.py

```diff
@@ -71,6 +71,7 @@ def testcombine_metadata_lists():
         (("a", "abc", None), ("a", ["abc"], [])),
         (("a", None, "abc"), ("a", [], ["abc"])),
         (("a", None, None), ("a", [], [])),
+        (("a", ("abc"), {"abc"}), ("a", ["abc"], ["abc"])),
```

Comment:
Actually we don't want to support sets here. Sets do not maintain order and most of the normalised features require strict ordering, e.g. partition on will behave differently when using differently ordered input

Reply:
Done, Please verify